src/ManageCourses.Api/Services/UserLogService.cs

```diff
@@ -26,7 +26,7 @@ public bool CreateOrUpdateUserLog(string signInUserId, string email)
 
                     var userLog = _context.UserLogs
                         .Include(x => x.User)
-                        .SingleOrDefault(x => (user != null ? x.User == user : true) && x.SignInUserId == signInUserId);
+                        .SingleOrDefault(x => (user != null ? x.User == user : true || !string.IsNullOrEmpty(x.UserEmail) ? string.Equals(x.UserEmail, email, StringComparison.InvariantCultureIgnoreCase) : true ) && x.SignInUserId == signInUserId);
```

Comment:
way too much nested logic: two ternary operators, a string equality comparison and two boolean operators! This should be made more readable

Reply:
I don't like that we duplicate the string comparison logic - can this be a method or extension method on McUser? (also to be used by `McUserQueryableExtensions.ByEmail`) cc. @timabell